azure/services/agentpools/agentpools.go

```diff
@@ -125,6 +125,13 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			},
 		}
 
+		// When autoscaling is set, the count of the nodes differ based on the autoscalar and should not depend on the
+		// count present in machinepool, azuremanagedmachinepool, hence we should not make an update api call based
+		// on difference in count.
+		if profile.EnableAutoScaling != nil && existingProfile.Count != nil {
```

Comment:
This seems completely reasonable to me. 

However, I wonder if we could add a condition or something to let the user know what is going on. There are sure to be questions like, "Why is my cluster scaled up so high when I gave it the desired state of X?". I could imagine a user getting quite irritated by CAPZ ignoring their desired state.

Reply:
+1

I agree we should try to surface something observable to the user to let them know that the actual count is different as autoscaling is enabled. I think this is the same thing that https://github.com/kubernetes-sigs/cluster-api/pull/5685 is trying to solve.

Reply:
also related to https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2022

we should try to solve the issue consistently across providers as much as possible to ensure a consistent user experience across CAPI providers

Reply:
After reading the above two threads, I feel along with the existing changes and the suggestions made above to surface something to the user to let know that the nodes have scaled up or scaled down because of the autoscaler we must also adjust the machinepool's desired replica count according to the node count present in cluster. 
In this way even the machinepools desired count and the phase spec will be accurate.